pool/src/lib.rs

```diff
@@ -688,6 +688,83 @@ impl LendingPoolTrait for LendingPool {
         Ok(())
     }
 
+    /// Enables or disables asset for using as collateral.
+    /// User should not have the debt in asset.
+    /// If user has debt position it will be checked if position stays good after disabling collateral.
+    ///
+    /// # Arguments
+    /// - who The address for collateral enabling/disabling
+    /// - asset The address of underlying asset
+    /// - use_as_collateral Enable/disable flag
+    ///
+    /// # Errors
+    /// - UserConfigNotExists
+    /// - NoReserveExistForAsset
+    /// - MustNotHaveDebt
+    /// - Bad position
+    ///
+    fn set_as_collateral(
+        env: Env,
+        who: Address,
+        asset: Address,
+        use_as_collateral: bool,
+    ) -> Result<(), Error> {
+        who.require_auth();
+        let mut user_config = read_user_config(&env, who.clone())?;
+        let reserve_index = read_reserve(&env, asset)?.get_id();
+        match (user_config.is_borrowing_any(), use_as_collateral) {
+            (true, true) => {
+                assert_with_error!(
+                    &env,
+                    !user_config.is_borrowing(&env, reserve_index),
+                    Error::MustNotHaveDebt
+                );
+                user_config.set_using_as_collateral(&env, reserve_index, use_as_collateral);
+                write_user_config(&env, who, &user_config);
+            }
+            (true, false)
+                if !user_config.is_borrowing(&env, reserve_index)
+                    && user_config.is_using_as_collateral(&env, reserve_index) =>
+            {
+                user_config.set_using_as_collateral(&env, reserve_index, use_as_collateral);
+                let reserves = read_reserves(&env);
+                let account_data = Self::calc_account_data(
+                    &env,
+                    who.clone(),
+                    None,
+                    &user_config,
+                    &reserves,
+                    false,
+                )?;
+                Self::require_good_position(&env, account_data);
+                write_user_config(&env, who, &user_config);
+            }
```

Comment:
Should we return `Error` in else statement since we do not write anything into the storage and return Ok as a result? The end-user would think they successfully updated the `asset` as not `use_as_collateral`.